src/setcpuaffinity.c

```diff
@@ -82,9 +87,9 @@ void setcpuaffinity(const char *cpulist) {
     if (!cpulist)
         return;
 
-#ifndef __NetBSD__
+#if defined(__linux__) || defined(__FreeBSD__)
     CPU_ZERO(&cpuset);
-#else
+#elif defined(__NetBSD__)
     cpuset = cpuset_create();
 #endif
```

Comment:
@devnexen i didn't dive too much into it (not really familiar with the code).
can you save me the effort and explain why MacOS isn't mentioned here?
and also at the very bottom of this file?

Reply:
Because there is no need in this place.

Reply:
so calling `thread_policy_set` in a loop appends to the affinity settings each time?
don't we at least need to reset the initial setting before starting to append to it?

Reply:
Yes it appends to the current thread but no need to reset. simpler api but in return less good granularity I feel compared to Linux.

Reply:
i'm sorry, i can't find any documentation that confirms this.
the API is named "Set" not "Append" and I also don't understand how it can append without first clearing the default.
I don't think this worth more time invested in it, and i rather not take the risk of this feature working on Mac by just using the last core rather than a set of cores (would rather that it'll be disabled and use all cores)

Reply:
No problem if you do not merge :-) it is indeed hard to find proper documentation for this sort of half hidden apis.